docs/scos/dev/guidelines/monitorable-process-guidelines.md

```diff
@@ -0,0 +1,86 @@
+* Scheduled tasks: all started scheduled tasks and accordingly their end-results (error, success, interrupt).
+* Application scoped resources: resource issues including exhausted resources, exceeded capacities and connectivity issues.
+* Threats: Suspicious/malicious activities against the application.
+* Unhandled/handled exceptions
+* Silenced exceptions: In case an empty “catch()” clause consumes the exception.
+* Significant user actions/events (including: help requests / cancelled actions)
+
+### What NOT to log?
+* Avoid logging unreasonably. When you decide to have a log-entry, you should have at least 1 use-case in mind where the related data 
+can be considered a helpful addition.
+* Avoid logging in performance critical places. In case there is a performance critical process or an unusually large cycle as a sub-process
+element, you should make a careful decision between performance vs traceability.
+* Audit trail (including authentication & authorization): it is recommended to log the audit trails of important entity changes but it should 
+not be sent to the regular log system since the typical data here consists mainly of sensitive data.
+
+### Log levels
+The following log levels MUST be utilised appropriately during application workflows to allow easy track down particular problems / investigations 
+in the System. A wrong category selection MAY de-rail a fellow inspector.
+
+
+| Level | Description |
+|-------|-------------|
+| Debug | Provides detailed data, which is mostly used for debugging. These logs allow to check variable values and/or error stacks. |
+| Info | Interesting events. |
+| Notice | Uncommon events. |
+| Warning | Describes events that are less destructive than errors. They usually do not result in any reduction of the program's functionality or its full failure. Undesirable things that are not necessarily wrong. |
+| Error | Identifies error events that may still let the software run, but with restricted capabilities in the impacted routes. |
+| Critical | Identifies extremely serious error events that are likely to cause the program to abort. Typically, this leads to catastrophic failures. |
+| Alert | Action must be taken immediately. |
+| Emergency | An urgent alert. |
+
+### Log entry format
+A log entry MUST always answer the following items and follow the below describes structure
+* **who** caused the event
+* **when** exactly did the event happened
+* **where** did the event took place (context, component, application, etc.)
+* **what** has happened and/or why has it happened
+* **result** of event (exception, success details, information details)
+
+**TODO: add code example blocks from https://spryker.atlassian.net/wiki/spaces/ALO/pages/3388014860/Monitorable+process+guidelines**
+
+## Metric generation
+Applies to all application components under ACI scoped processes.
+
+Each metric represents a condition of some system attribute. You can have a lot of them and correlate them with each other.
```

Comment:
Is there a guidelines on how metrics can be tracked on the project - any library we suggest?